src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IIsPatternExpression.cs

```diff
@@ -1182,6 +1182,279 @@ void M1(object o, bool b)
             VerifyOperationTreeForTest<IsPatternExpressionSyntax>(compilation, expectedOperationTree);
         }
 
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Patterns)]
+        [Fact]
+        public void IsPattern_BadRecursivePattern_04()
+        {
+            var source = @"
+class C
+{
+    void M1(object o)
+    {
+        _ = /*<bind>*/o is D { A: var a }/*</bind>*/;
+    }
+}
+class D
+{
+    public event System.Action A;
+}
+";
+
+            var compilation = CreateCompilation(source);
+            compilation.VerifyDiagnostics(
+                // (6,32): error CS0154: The property or indexer 'A' cannot be used in this context because it lacks the get accessor
```

Comment:
>The property or indexer 'A' cannot be used in this context because it lacks the get accessor [](start = 41, length = 92)

The error feels misleading, E is not a property and is not an indexer.

Reply:
Please file an issue about that. It is not relevant to any compiler changes in this PR.

---
In reply to: [449141336](https://github.com/dotnet/roslyn/pull/45454#discussion_r449141336) [](ancestors = 449141336)